docs/httplug.md

```diff
@@ -0,0 +1,52 @@
+# Httplug HTTP client abstraction
+
+Httplug is an abstraction for HTTP clients. There are two main use cases:
+
+1. Usage in a project
+2. Usage in a reusable package
+
+In both cases, the client provides a `sendRequest` method to send a PSR-7 `RequestInterface` and returns a PSR-7 `ResponseInterface` or throws an exception that implements `Http\Client\Exception`. The client also has a `sendRequests` to send requests in parallel. Clients that do not support sending requests in parallel will just send them sequentially.
+
+See the (tutorial)[tutorial.md] for a concrete example.
+
+
+## Httplug implementations
+
+Httplug implementations typically are either HTTP clients of their own, or they are adapters wrapping existing clients like Guzzle 6. In the later case, they will depend on the required client implementation, you should only need to require the adapter and not the actual client.
+
+Clients usually also provide a (message factory)[message-factory.md] to allow you to create requests. See (packagist)[https://packagist.org/providers/php-http/client-implementation] for the full list of implementations.
+
+Note: Until Httplug 1.0 becomes stable, we will focus on the Guzzle6 adapter.
+
+## Usage in a project
+
+When installing in an application or a non-reusable package, you should require a concrete client implementation. The client will in turn depend on `php-http/httplug`.
```

Comment:
I am not sure referring to packages by putting them between backticks is the way we should do it (although I am the one who brought this into practice). Options:

1. Leave it as is
2. Link to github repo
3. Link to packagist